### PR TITLE
STASH-7800 If cookie-secure is false, mark the Hazelcast session cookie as "Secure" if the request that initiated the corresponding session used HTTPS

### PR DESCRIPTION
**Pull request by cszmajda**

### CODE REVIEW

File: hazelcast-wm/src/main/java/com/hazelcast/web/WebFilter.java [hunk 1]
```diff
@@ -78,8 +78,10 @@
  * <li>{@code cookie-name}: Sets the name for the Hazelcast session cookie (Default:
  * {@link #HAZELCAST_SESSION_COOKIE_NAME "hazelcast.sessionId"}</li>
  * <li>{@code cookie-domain}: Sets the domain for the Hazelcast session cookie (Default: {@code null})</li>
- * <li>{@code cookie-secure}: When enabled, indicates the Hazelcast session cookie should only be sent over
- * secure protocols (Default: {@code false})</li>
+ * <li>{@code cookie-secure}: When set to {@code true}, indicates the Hazelcast session cookie should always be marked
+ * "Secure", even if the request that initiated the corresponding session is using plain HTTP instead of HTTPS. If
+ * {@code false}, it will be marked as secure only if the request that initiated the corresponding session was also
+ * secure. (Default: {@code false})</li>
```
Comment (cszmajda) [inline hazelcast-wm/src/main/java/com/hazelcast/web/WebFilter.java]: This handling of the `cookie-secure` boolean conforms to the Java EE Servlet 3.0 API
http://docs.oracle.com/javaee/7/api/javax/servlet/SessionCookieConfig.html#isSecure--.

**Comment by cszmajda** [comment]
@bturner @mheemskerk @juretta This is the PR against hazelcast to prevent it from returning in insecure cookie when the incoming request is HTTPS.

**Comment by mheemskerk** [comment]
LGTM

**Comment by juretta** [comment]
Thanks for fixing this up Chris.

**Pull request merged**